### PR TITLE
test/e2e/libvirt: UploadPodvm() Go Rewrite

### PR DESCRIPTION
**Pull request by bookinabox**
Rewrites the UploadPodvm() implementaiton in for libvirt testing in Go rather than a call to virsh.

Fixes: #1162

### CODE REVIEW

**Comment by wainersm** [comment]
Hi @bookinabox , I ran the e2e libvirt suite with this changes and it failed all. I suspect the podvm image is copied but corrupted. I will find some time to debug it.

File: test/provisioner/provision_libvirt.go [hunk 1]
```diff
@@ -179,36 +179,31 @@ func (l *LibvirtProvisioner) GetProperties(ctx context.Context, cfg *envconf.Con
...
+		return err
+	}
+	length := uint64(file.Size())
+
+	sVol, err := sPool.LookupStorageVolByName(l.volumeName)
+	if err != nil {
+		return err
+	}
+
+	stream, err := l.conn.NewStream(libvirt.STREAM_NONBLOCK)
```
Comment (wainersm) [inline test/provisioner/provision_libvirt.go:198]: Hi @bookinabox , I was looking at the libvirt API. It seems we will need a blocking stream as well as cancel/abort/finish the stream properly. See https://libvirt.org/html/libvirt-libvirt-stream.html#virStreamNew

Comment (bookinabox) [reply test/provisioner/provision_libvirt.go]: Fixed and tested!

**Comment by wainersm** [comment]
just an update on this PR...there is some bug on the code still and @bookinabox is working to fix.

**Comment by bookinabox** [comment]
Should be done! Didn't realize I needed to sendall from the stream after until I saw this: 	// https://github.com/libvirt/libvirt-go/blob/master/integration_test.go#L1484C1-L1498C3

**Comment by wainersm** [comment]
> Should be done! Didn't realize I needed to sendall from the stream after until I saw this: // https://github.com/libvirt/libvirt-go/blob/master/integration_test.go#L1484C1-L1498C3

Cool. Never worked with the stream API of libvirt, so I learned something today :)

Could you please fix the warnings raised by golangci-lint? -> https://github.com/confidential-containers/cloud-api-adaptor/pull/1191/commits/efc918c08d848a73b31b2bb4348bceb42466bf0e

**Comment by wainersm** [comment]
@bookinabox now we are talking :)

```
Wait for the cc-operator-daemon-install DaemonSet be available
Wait for the pod cc-operator-daemon-install-mtjjn be ready
Wait for the cloud-api-adaptor-daemonset DaemonSet be available
Wait for the pod cloud-api-adaptor-daemonset-nhggq be ready
Wait for the kata-remote runtimeclass be created
=== RUN   TestLibvirtCreateSimplePod
=== RUN   TestLibvirtCreateSimplePod/SimplePeerPod_test
=== RUN   TestLibvirtCreateSimplePod/SimplePeerPod_test/PodVM_is_created
time="2023-07-27T16:55:17-03:00" level=info msg="Deleting pod... nginx"
--- PASS: TestLibvirtCreateSimplePod (115.06s)
    --- PASS: TestLibvirtCreateSimplePod/SimplePeerPod_test (115.06s)
        --- PASS: TestLibvirtCreateSimplePod/SimplePeerPod_test/PodVM_is_created (0.04s)
=== RUN   TestLibvirtCreatePodWithConfigMap
=== RUN   TestLibvirtCreatePodWithConfigMap/ConfigMapPeerPod_test
=== RUN   TestLibvirtCreatePodWithConfigMap/ConfigMapPeerPod_test/Configmap_is_created_and_contains_data
time="2023-07-27T16:56:42-03:00" level=info msg="Data Inside Configmap: Hello, world"
time="2023-07-27T16:56:42-03:00" level=info msg="Deleting Configmap... nginx-config"
time="2023-07-27T16:56:42-03:00" level=info msg="Deleting pod... nginx-configmap-pod"
--- PASS: TestLibvirtCreatePodWithConfigMap (85.12s)
    --- PASS: TestLibvirtCreatePodWithConfigMap/ConfigMapPeerPod_test (85.12s)
        --- PASS: TestLibvirtCreatePodWithConfigMap/ConfigMapPeerPod_test/Configmap_is_created_and_contains_data (5.08s)
=== RUN   TestLibvirtCreatePodWithSecret
=== RUN   TestLibvirtCreatePodWithSecret/SecretPeerPod_test
=== RUN   TestLibvirtCreatePodWithSecret/SecretPeerPod_test/Secret_has_been_created_and_contains_data
time="2023-07-27T16:57:57-03:00" level=info msg="Username from secret inside pod: admin"
time="2023-07-27T16:58:02-03:00" level=info msg="Password from secret inside pod: password"
time="2023-07-27T16:58:02-03:00" level=info msg="Deleting Secret... nginx-secret"
time="2023-07-27T16:58:02-03:00" level=info msg="Deleting pod... nginx-secret-pod"
--- PASS: TestLibvirtCreatePodWithSecret (80.16s)
    --- PASS: TestLibvirtCreatePodWithSecret/SecretPeerPod_test (80.16s)
        --- PASS: TestLibvirtCreatePodWithSecret/SecretPeerPod_test/Secret_has_been_created_and_contains_data (10.11s)
PASS
ok      github.com/confidential-containers/cloud-api-adaptor/test/e2e   1608.047s
```

**Review by wainersm — APPROVED** [review]
Thanks @bookinabox !

**Review by snir911 — APPROVED** [review]
LGTM thanks!

**Pull request merged**